### ISSUE TITLE
Should SRP0009 be aplied only on non-inlineable UDFs?

### ISSUE THREAD

**Issue by JocaPC** [ErikEJ/SqlServer.Rules#159]
I'm getting this warning when I use system YEAR() function in the where clause:

Warning SRP0009 : SqlServer.Rules : Avoid wrapping columns within a function
in the WHERE clause. (Sargable) 
(https://github.com/ErikEJ/SqlServer.Rules/blob/master/docs/Performance/SRP0009.md)

Shoudl this rule work only for user defined functions or both for user defined and system functions. The user defined functions in WHERe might be a problem but not system functions. Also if there is a way to identify that function is inlinable maybe we don't need this warning because inline UDF should have good perf.

```sql
CREATE VIEW [dbo].[vw_PaymentAnalysis] AS
SELECT
    T.PaymentType
    ,COUNT(T.PaymentType) AS PaymentsCount
    ,SUM(T.TotalAmount) AS TotalAmountProcessed
FROM dbo.Trip AS T
JOIN dbo.[Date] AS D
    ON T.[DateID]=D.[DateID]
WHERE YEAR(D.[Date])=2013
GROUP BY T.PaymentType
```

**Comment by ErikEJ** [comment]
Would that not require a calculated column in an index to.perform well?

**Comment by JocaPC** [comment]
That's a hard question. 
`WHERE f(column) = value` might be fast if `f` is a built-in function, inlineable user defined scalar funcction, user defined function with a matching computed column+index, but it will be slow if it is non-inlineable user defined function without computed column+index. also it might be slow if it is heavy built in function like spatial or json.

I don't know do you really want to send a generic warning at compile time, because I think that more people would be surprised what is the error. To be more reliable you would need to do many metadata inspections.

**Comment by JocaPC** [comment]
Yes, I would remove it. When I look at the query below I could not find alternative if I need to filter by year.
So if the warning would not influence the owner of this query to change it (because there is no alternative), it is just a hypothetical warning.

SELECT
    T.PaymentType
    ,COUNT(T.PaymentType) AS PaymentsCount
    ,SUM(T.TotalAmount) AS TotalAmountProcessed
FROM dbo.Trip AS T
JOIN dbo.[Date] AS D
    ON T.[DateID]=D.[DateID]
WHERE YEAR(D.[Date])=2013
GROUP BY T.PaymentType

**Comment by ErikEJ** [comment]
I will keep the rule, and perhaps improve the docs. If users do not like it, they can disable
